CMakePresets.json

```diff
@@ -0,0 +1,125 @@
+            "generator": "Ninja Multi-Config"
+        },
+        {
+            "name": "debug",
+            "displayName": "Debug",
+            "description": "Debug build with make.",
+            "inherits": "defaults",
+            "generator": "Unix Makefiles",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Debug"
+            }
+        },
+        {
+            "name": "devel",
+            "displayName": "Devel",
+            "description": "Developer build using make.",
+            "inherits": "defaults",
+            "generator": "Unix Makefiles",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Devel"
+            }
+        },
+        {
+            "name": "release",
+            "displayName": "Release",
+            "description": "Release build using make.",
+            "inherits": "defaults",
+            "generator": "Unix Makefiles",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release"
+            }
+        },
+        {
+            "name": "debug-ninja",
+            "displayName": "Debug Ninja",
+            "description": "Debug build using ninja.",
+            "inherits": "defaults",
+            "generator": "Ninja",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Debug"
+            }
+        },
+        {
+            "name": "devel-ninja",
+            "displayName": "Devel Ninja",
+            "description": "Developer build using ninja.",
+            "inherits": "defaults",
+            "generator": "Ninja",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Devel"
+            }
+        },
+        {
+            "name": "release-ninja",
+            "displayName": "Release Ninja",
+            "description": "Release build using ninja.",
+            "inherits": "defaults",
+            "generator": "Ninja",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release"
+            }
+        },
+        {
+            "name": "clang-debug",
+            "displayName": "Clang Debug",
+            "description": "Debug build using ninja and clang.",
+            "generator": "Ninja",
+            "inherits": "clang",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Debug"
+            }
+        },
+        {
+            "name": "clang-devel",
+            "displayName": "Clang Devel",
+            "description": "Developer build using ninja & clang.",
+            "generator": "Ninja",
+            "inherits": "clang",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Devel"
+            }
+        },
+        {
+            "name": "clang-release",
+            "displayName": "Clang Release",
+            "description": "Release build using ninja & clang.",
+            "generator": "Ninja",
+            "inherits": "clang",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release"
```

Comment:
It's worthwhile enabling LTO here , which is usually `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON`. Although ideally you'd want to use lld as well (`-fuse-ld=lld`), the link times with the gcc linker are pretty terrible.

Reply:
All doable. The questions would be whether I should enable LTO on all the presets, just some, or add new presets for it. Largely just if none of the presets don't have LTO, not using it will be less tested.

With ld, the main thing would be that I don't want to assume people have it installed (which is really the only reason I included non-ninja builds). It is possible to do conditionals, but I'd need something I could check (a cmake variable or environment variable) to see if it's installed, though I suppose I can assume it's installed on the clang builds.

I'd rather do conditionals on ninja, too, really, but I was having trouble finding something I could use, and I figured it was something I could refine later...